README.md

```diff
@@ -49,6 +49,51 @@ Here is the guide:
 
 * [Add key to Github](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)
 
+### [IMPORTANT!] [WSL USERS] To ensure that WSL shares the SSH keys with the devContainer perform:
+
+#### 1. Create a file named ssh-agent in your .ssh directory
+```bash
+touch ~/.ssh/ssh-agent
+```
+
+#### 2. Check the name of your shell config file name. Expect `.profile` or `.bash_profile`
+
+```bash
+cd ~ && ls -a
+```
+
+>In the list that appears see if your system has a .profile or a .bash_profile
+
+#### 3. Add the following block of code in the end of your shell config file.
+
+```bash
+# Auto-start ssh-agent
+if [ -z "$SSH_AUTH_SOCK" ]; then
+  # Check for a currently running instance of the agent
+  RUNNING_AGENT="`ps -ax | grep 'ssh-agent -s' | grep -v grep | wc -l | tr -d '[:space:]'`"
+  if [ "$RUNNING_AGENT" = "0" ]; then
+    # Launch a new instance of the agent
+    ssh-agent -s &> ~/.ssh/ssh-agent
+  fi
+  eval `cat ~/.ssh/ssh-agent` > /dev/null
+  ssh-add ~/.ssh/id_ed25519 2> /dev/null
+fi
+```
+#### Open your shell config file and add the above code in the end:
+
+```bash
+nano ~/.profile
+```
+#### OR
+```bash
+nano ~/.bash_profile
+```
+Depending on how the file is named in your system
+
+#### 4. After that `Save & Exit`
+
+#### 5. Then reopen WSL
+
```

Comment:
I suggest this thing as a whole to be extracted as a script. Which they should run after adding the ssh keys to their GitHub profiles and cloning the project. (1. Add shh to github, 2. clone project, 3. Run script) And after the script has finished to prompt them to restart their shell via `exec $SHELL`

Reply:
Yes!

Reply:
I would also suggest putting a coment with the link to the official docs if people are interested https://code.visualstudio.com/remote/advancedcontainers/sharing-git-credentials

And if you can find an article explaining how this ssh-agent frowarding happens, because it's managed by the VS code extension. From what I read you have an ENV var called `SSH_AUTH_SOCKET`  which is mapped from the host to the container, and that's how the container can access the ssh-agent of the host.
It's good stuff.

Reply:
Yes sir! That is why you need to make sure you have the ssh-agent running and your keys added to this agent. I think this process creates the `SSH_AUTH_SOCKET`.

Reply:
The script above just checks the value of the env variable. It is expected to be created beforehand.

Reply:
Yeah correct! I should have said updates its value if its empty

Reply:
Resolves 545fec6